pkg/analyzer/detection/detection.go

```diff
@@ -15,15 +16,27 @@ import (
 // LogLayerName is the name of the log field holding the detection target.
 const LogLayerName = "layer"
 
+// DetectComponentOpts contains configurations how components detection works
+type DetectComponentOpts struct {
+	// UncertifiedRHEL is boolean to decide if OS is uncertified RHEL
+	UncertifiedRHEL bool
+
+	// IsRHCOSRequired: if CoreOS is required for DetectComponents, Node scanning is disabled if false
+	IsRHCOSRequired bool
+}
+
 // DetectComponents detects the namespace and extracts the components present in
 // the files of a filesystem or image layer. For layers, the parent layer should
 // be specified. For filesystems, which don't have the concept of intermediate
 // layers, or the root layer, use `nil`. Notice that language components are not
 // extracted by DetectComponents, but if provided they are annotated with
 // certified RHEL dependencies, and returned.
-func DetectComponents(name string, files analyzer.Files, parent *database.Layer, languageComponents []*component.Component, uncertifiedRHEL bool) (*database.Namespace, []database.FeatureVersion, *database.RHELv2Components, []*component.Component, error) {
-	namespace := DetectNamespace(name, files, parent, uncertifiedRHEL)
-
+func DetectComponents(name string, files analyzer.Files, parent *database.Layer, languageComponents []*component.Component, opts DetectComponentOpts) (*database.Namespace, []database.FeatureVersion, *database.RHELv2Components, []*component.Component, error) {
```

Comment:
Nit: Should we maybe update the comment above to include the info about the early exit on non-RHCOS namespaces if the scan opts are set?